sys/androidmedia/gstamcvideodec.c

```diff
@@ -1697,10 +1675,20 @@ gst_amc_video_dec_set_format (GstVideoDecoder * decoder,
 
 #if USE_AMCVIDEOSINK
     if (klass->direct_rendering && self->surface == NULL) {
+      GstCaps *caps;
+      GstBuffer *buf;
+
       /* Exposes pads with decodebin with a dummy buffer to link with the sink
        * and get the surface */
       GST_INFO_OBJECT (self, "Sending a dummy buffer");
-      gst_amc_video_dec_push_dummy (self, TRUE);
```

Comment:
In my opinion it was more clear letting this code inside gst_amc_video_dec_push_dummy().